lxd/db/query/slices_test.go

```diff
@@ -0,0 +1,73 @@
+package query_test
+
+import (
+	"database/sql"
+	"fmt"
+	"testing"
+
+	_ "github.com/mattn/go-sqlite3"
+	"github.com/mpvl/subtest"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/lxc/lxd/lxd/db/query"
+)
+
+// Exercise possible failure modes.
+func TestStrings_Error(t *testing.T) {
+	for _, c := range testStringsErrorCases {
+		subtest.Run(t, c.query, func(t *testing.T) {
+			tx := newTxForSlices()
+			values, err := query.Strings(tx, c.query)
+			assert.EqualError(t, err, c.error)
+			assert.Nil(t, values)
+		})
+	}
+}
+
+// All values yield by the query are returned.
+func TestStrings(t *testing.T) {
+	tx := newTxForSlices()
+	values, err := query.Strings(tx, "SELECT name FROM test ORDER BY name")
+	assert.Nil(t, err)
+	assert.Equal(t, []string{"bar", "foo"}, values)
+}
+
+// Exercise possible failure modes.
+func TestIntegers_Error(t *testing.T) {
+	for _, c := range testIntegersErrorCases {
+		subtest.Run(t, c.query, func(t *testing.T) {
+			tx := newTxForSlices()
+			values, err := query.Integers(tx, c.query)
+			assert.EqualError(t, err, c.error)
+			assert.Nil(t, values)
+		})
+	}
+}
+
+// All values yield by the query are returned.
+func TestIntegers(t *testing.T) {
+	tx := newTxForSlices()
+	values, err := query.Integers(tx, "SELECT id FROM test ORDER BY id")
+	assert.Nil(t, err)
+	assert.Equal(t, []int{0, 1}, values)
+}
+
+// Return a new transaction against an in-memory SQLite database with a single
+// test table populated with a few rows.
+func newTxForSlices() *sql.Tx {
+	db, err := sql.Open("sqlite3", ":memory:")
+	if err != nil {
+		panic(fmt.Sprintf("failed to create database: %v", err))
+	}
+	if _, err := db.Exec("CREATE TABLE test (id INTEGER, name TEXT)"); err != nil {
+		panic(fmt.Sprintf("failed to create table: %v", err))
+	}
+	if _, err := db.Exec("INSERT INTO test VALUES (0, 'foo'), (1, 'bar')"); err != nil {
+		panic(fmt.Sprintf("failed to insert rows: %v", err))
+	}
+	tx, err := db.Begin()
+	if err != nil {
+		panic(fmt.Sprintf("failed to begin transaction: %v", err))
+	}
+	return tx
```

Comment:
I realize this is in test code, but the thought of having some code in the daemon code tree which calls "panic()" makes me uncomfortable, I don't want people to think that it's ever okay to call panic() from the LXD daemon.